app/components/channel_modal.js

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {Component, PropTypes} from 'react';
+import {Animated, Dimensions} from 'react-native';
+
+const AnimatedView = Animated.View;
+const {width: deviceWidth, height: deviceHeight} = Dimensions.get('window');
+
+export default class ChannelModel extends Component {
+    static propTypes = {
+        children: PropTypes.node,
+        duration: PropTypes.number,
+        topOffset: PropTypes.number,
+        theme: React.PropTypes.object.isRequired,
+        visible: PropTypes.bool.isRequired
+    }
+
+    static defaultProps = {
+        duration: 400,
+        topOffset: 0,
+        visible: false
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            top: new Animated.Value(deviceHeight)
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.visible === nextProps.visible) {
+            return;
+        }
+
+        const animateValue = nextProps.visible ? this.props.topOffset : deviceHeight;
+
+        Animated.timing(this.state.top, {
+            toValue: animateValue,
+            duration: this.props.duration
+        }).start();
+    }
+
+    render() {
+        return (
+            <AnimatedView style={{flex: 1, position: 'absolute', top: this.state.top, left: 0, height: (deviceHeight - this.props.topOffset), width: deviceWidth, backgroundColor: '#000', opacity: 0.95, overflow: 'hidden'}}>
```

Comment:
@enahum, @hmhealey, @jarredwitt 
Because of the `no-magic-numbers` rule, eslint is complaining about this line (and about most every style set we have). I'd rather not ditch the rule completely; it has value in some contexts (see comment above), but it's gonna be more of a nuisance in our styles. This might be further reason to move these styles into an embedded stylesheet. That way we can ignore the eslint rule from there to the end of the file. At least I hope we can do this. I'd prefer to avoid having to disable the rule inline in a block of jsx or to move the styles to their own file simply for the sake of disabling eslint there.